File: java_to_ruby/SocketAccess.java
// This code is excerpted from http://www.slideshare.net/uehaj/dev-love-groovy2011 p.7
// written by @uehaj

import java.io.*;
import java.net.*;

public class SocketAccess {
    public static void main(String[] args) {
        Socket soc = null;
        InputStream ins = null;
        OutputStream outs = null;
        try {
            soc = new Socket("www.java-users.jp", 80);
            ins = soc.getInputStream();
            outs = soc.getOutputStream();
            outs.write("GET / HTTP/1.0\n\n".getBytes());
            BufferedReader bis = new BufferedReader(new InputStreamReader(ins));
            String line;
            while ((line = bis.readLine()) != null) {
                System.out.println(line);
            }
        }
        catch(IOException e) {
            e.printStackTrace();
            try {
                if (soc != null) soc.close();
            }
            catch(IOException ex) {}
        }
    }
}

File: jjug-jruby-2011/java_to_ruby/socket_access.rb
require 'java'

java_import 'java.io.BufferedReader'
java_import 'java.io.InputStreamReader'
java_import 'java.io.IOException'
java_import 'java.net.Socket'

begin
  soc = Socket.new("www.java-users.jp", 80)
  soc.output_stream.write("GET / HTTP/1.0\r\n\r\n".to_java.bytes)
  bis = BufferedReader.new(InputStreamReader.new(soc.input_stream))
  while line = bis.read_line
    puts line
  end
rescue IOException => e
  puts e.backtrace
  begin
    soc.close if soc
  rescue IOException
  end
end
